api/v1alpha1/shared_types.go

```diff
@@ -92,6 +105,36 @@ type KubernetesDeploymentSpec struct {
 	// TODO: Expose config as use cases are better understood, e.g. labels.
 }
 
+// KubernetesDaemonsetSpec defines the desired state of the Kubernetes daemonset resource.
+type KubernetesDaemonSetSpec struct {
+	// Patch defines how to perform the patch operation to daemonset
+	//
+	// +optional
+	Patch *KubernetesPatchSpec `json:"patch,omitempty"`
+
+	// The daemonset strategy to use to replace existing pods with new ones.
```

Comment:
for your use case, do you need all these fields - pod, container, initcontainers ?

Reply:
In my case I test many cases, but most important:
```
kind: EnvoyProxy
metadata:
  name: envoy-proxy-config
  namespace: envoy-gateway-system
spec:
  provider:
    type: Kubernetes
    kubernetes:
      mode: DaemonSet
      envoyService:
        disable: true
      envoyDaemonSet:
        pod:
          hostNetwork: true
          tolerations:
          - key: node-role.kubernetes.io/control-plane
            operator: Exists
          nodeSelector:
            node-role.kubernetes.io/control-plane: ""
```

Start Envoy Proxy like daemonset on special nodes with hostnetwork=true

May be for another cases someone need this?